src/bin/mlem.rs

```diff
@@ -120,7 +120,9 @@ fn main() -> Result<(), Box<dyn Error>> {
         for (n, image) in (Image::mlem(vbox, &measured_lors, args.tof, args.cutoff, &sensitivity_matrix))
             .take(args.iterations)
             .enumerate() {
-                report_time("iteration");
+                println!("--------------------------");
+                println!("Iteration number {}", n);
+                report_time("Time");
```

Comment:
This makes the output twice as long (4 lines instead of 2, per iteration). I'm not too keen on that.

Why don't you inject the iteration number into the `report_time` message? You can do that using `format!`, of which you can find an example a few lines lower down.

If you really want to have a separator line between the iterations, then consider a blank line rather than `-------------`, which tends to be even easier for the human eye no understand at a glance.

Reply:
> Why don't you inject the iteration number into the report_time message? You can do that using format!, of which you can find an example a few lines lower down.

Yeah, I thought about this, but `report_time` is also used in line 96: `Loaded LOR data from local disk` which is not inside the iteration loop, and also I didn't want to print more than once the iteration number.

> If you really want to have a separator line between the iterations, then consider a blank line rather than -------------, which tends to be even easier for the human eye no understand at a glance.

I see, I'll do it!

Reply:
if I do: `report_time(format!("Iteration number {}", n));`, I get the error: `expected `&str`, found struct `String``

Reply:
Hmm, yes, how unfortunate. This is a case where Rust's care and precision makes something trivial too difficult. And there's also some subtlety with the type inference in the closure, that I don't fully understand.

I suggest you 

1. explicitly specify the parameter type of the `report_time` closure as `&str` (on line 87): 
   ```rust
   let mut report_time = |message: &str| {
   ```

2. give `report_time` a reference to the string that you make with `format!`, by adding a `&` between `report_time(` and `format!`.

Reply:
Now it works! And I think since there are only two lines for every iteration, there is no need to separate the iterations.